akka-http2-support/src/main/scala/akka/http/impl/engine/http2/framing/Http2FrameParsing.scala

```diff
@@ -10,15 +10,36 @@ import akka.stream.Attributes
 import akka.stream.impl.io.ByteStringParser
 import akka.stream.stage.GraphStageLogic
 import Http2Protocol.FrameType._
-import Http2Protocol.{ ErrorCode, Flags, FrameType, SettingIdentifier }
+import Http2Protocol.{ErrorCode, Flags, FrameType, SettingIdentifier}
 import akka.annotation.InternalApi
-
 import FrameEvent._
 
+import scala.annotation.tailrec
+
+/** INTERNAL API */
+@InternalApi
+private[http] object Http2FrameParsing {
+
+  def readSettings(payload: ByteStringParser.ByteReader): immutable.Seq[Setting] = {
+    @tailrec def readSettings(read: List[Setting]): immutable.Seq[Setting] =
+      if (payload.hasRemaining) {
+        val id = payload.readShortBE()
+        val value = payload.readIntBE()
+        if (isKnownId(id)) readSettings(Setting(SettingIdentifier.byId(id), value) :: read)
+        else readSettings(read)
+      } else read.reverse
+
+    readSettings(Nil)
+  }
```

Comment:
Needed to be opened up to be able to parse the settings frame out of the `Http2-Settings` header that comes with the upgrade